### PR TITLE
Update README: add `static_multimap` section and godbolt live examples

### PR DESCRIPTION
**Pull request by PointKernel**
This PR adds `static_multimap` section in README and provides godbolt links to all examples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,16 +183,25 @@ We plan to add many GPU-accelerated, concurrent data structures to `cuCollection
...
 ### `static_map`
 
-`cuco::static_map` is a fixed-size hash table using open addressing with linear probing. 
+`cuco::static_map` is a fixed-size hash table using open addressing with linear probing. See the Doxygen documentation in `static_map.cuh` for more detailed information.
 
-It provides both host, bulk APIs ([example](https://github.com/NVIDIA/cuCollections/blob/dev/examples/static_map/host_bulk_example.cu)) as well as device APIs for individual operations ([example](https://github.com/NVIDIA/cuCollections/blob/dev/examples/static_map/device_view_example.cu)).
+#### Examples:
+- [Host-bulk APIs](https://github.com/NVIDIA/cuCollections/blob/dev/examples/static_map/host_bulk_example.cu) (see [live example in godbolt](https://godbolt.org/z/ervPzqh64))
+- [Device-view APIs for individual operations](https://github.com/NVIDIA/cuCollections/blob/dev/examples/static_map/device_view_example.cu) (see [live example in godbolt](https://godbolt.org/z/qMWrfE6ET))
+- [Custom date type, key equality and hash function](https://github.com/NVIDIA/cuCollections/blob/dev/examples/static_map/custom_type_example.cu) (see [live example in godbolt](https://godbolt.org/z/oGfYjzMGT))
```
Comment (sleeepyjack) [inline README.md]:
```suggestion
- [Custom data types, key equality operators and hash functions](https://github.com/NVIDIA/cuCollections/blob/dev/examples/static_map/custom_type_example.cu) (see [live example in godbolt](https://godbolt.org/z/oGfYjzMGT))
```

File: README.md [hunk 2]
```diff
@@ -183,16 +183,25 @@ We plan to add many GPU-accelerated, concurrent data structures to `cuCollection
...
 
-See the Doxygen documentation in `static_map.cuh` for more detailed information.
+### `static_multimap`
+
+`cuco::static_multimap` is a fixed-size hash table that supports equivalent keys. It uses double hashing by default and supports switching to linear probing. See the Doxygen documentation in `static_multimap.cuh` for more detailed information.
```
Comment (sleeepyjack) [inline README.md]:
```suggestion
`cuco::static_multimap` is a fixed-size hash table that supports storing equivalent keys. It uses double hashing by default and supports switching to linear probing. See the Doxygen documentation in `static_multimap.cuh` for more detailed information.
```

File: README.md [hunk 3]
```diff
@@ -183,16 +183,25 @@ We plan to add many GPU-accelerated, concurrent data structures to `cuCollection
...
+
+#### Examples:
+- [Host-bulk APIs](https://github.com/NVIDIA/cuCollections/blob/dev/examples/static_multimap/host_bulk_example.cu) (see [live example in godbolt](https://godbolt.org/z/Po4eTEn1a))
 
 ### `dynamic_map`
 
-`cuco::dynamic_map` links together multiple `cuco::static_map`s to provide a hash table that can grow as keys are inserted. 
+`cuco::dynamic_map` links together multiple `cuco::static_map`s to provide a hash table that can grow as keys are inserted. It currently only provides host-bulk APIs. See the Doxygen documentation in `dynamic_map.cuh` for more detailed information.
```
Comment (sleeepyjack) [inline README.md]:
```suggestion
`cuco::dynamic_map` links together multiple `cuco::static_map`s to provide a hash table that can grow as key-value pairs are inserted. It currently only provides host-bulk APIs. See the Doxygen documentation in `dynamic_map.cuh` for more detailed information.
```

File: README.md [hunk 4]
```diff
@@ -183,16 +183,25 @@ We plan to add many GPU-accelerated, concurrent data structures to `cuCollection
...
+
+#### Examples:
+- [Host-bulk APIs (TODO)]()
```
Comment (sleeepyjack) [inline README.md:205]: open TODO

File: examples/static_map/device_view_example.cu [hunk 5]
```diff
@@ -28,7 +28,7 @@
 #include <limits>
 
 /**
- * @file device_side_example.cu
+ * @file device_view_example.cu
```
Comment (sleeepyjack) [inline examples/static_map/device_view_example.cu:31]: Can we make the CI/`pre-commit` doxygen check aware for this kind of thing?

**Review by sleeepyjack — CHANGES_REQUESTED** [review]
Thanks @PointKernel for adding valuable content to the project's landing page!

While scrolling through the readme, I was reminded that the doxygen docs page is still a TODO. Let me check if we can abuse the CI to automatically build the docs and push to the project's GH.io page.

**Comment by sleeepyjack** [comment]
> While scrolling through the readme, I was reminded that the doxygen docs page is still a TODO. Let me check if we can abuse the CI to automatically build the docs and push to the project's GH.io page.

We can either utilize our beloved gpuCI with a custom script or GH actions: https://github.com/marketplace/actions/doxygen-action + https://github.com/peaceiris/actions-gh-pages

File: examples/static_map/device_view_example.cu (hunk 5 above)
Comment (PointKernel) [reply examples/static_map/device_view_example.cu:31]: Spent some time looking into this and find nothing useful.

File: README.md (hunk 4 above)
Comment (PointKernel) [reply README.md:205]: Resolving this since the dynamic map example is not added yet.

**Review by sleeepyjack — APPROVED** [review]
LGTM

**Review by jrhemstad — APPROVED** [review]

**Pull request merged**